src/components/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,864 @@
+              <KButton
+                v-for="(timeFrame, itemIdx) in item.values"
+                :key="`time-${itemIdx}`"
+                :is-rounded="false"
+                :class="{ 'selected-option': timeFrame.key === selectedTimeframe.key }"
+                :data-testid="'select-timeframe-' + timeFrame.timeframeLength()"
+                class="timeframe-btn"
+                appearance="outline"
+                size="medium"
+                @click="changeRelativeTimeframe(timeFrame)"
+              >
+                {{ ucWord(timeFrame.timeframeText) }}
+              </KButton>
+            </div>
+          </div>
+        </div>
+      </template>
+      <template
+        v-if="!hidePopover"
+        #footer
+      >
+        <div class="d-flex justify-content-end">
+          <KButton
+            :is-rounded="false"
+            data-testid="k-datetime-picker-clear"
+            class="action-btn"
+            size="medium"
+            appearance="btn-link"
+            @click="clearSelection()"
+          >
+            Clear
+          </KButton>
+          <KButton
+            :disabled="submitDisabled"
+            :is-rounded="false"
+            data-testid="k-datetime-picker-submit"
+            class="action-btn"
+            size="medium"
+            appearance="btn-link"
+            @click="submitTimeFrame()"
+          >
+            Apply
+          </KButton>
+        </div>
+      </template>
+    </KPop>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, nextTick, onMounted, PropType, reactive, ref, toRefs, watch } from 'vue'
+import { format } from 'date-fns'
+import { DatePicker } from 'v-calendar'
+import KButton from '@/components/KButton/KButton.vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import KPop from '@/components/KPop/KPop.vue'
+import KSegmentedControl from '@/components/KSegmentedControl/KSegmentedControl.vue'
+import 'v-calendar/dist/style.css'
+
+export interface TimeRange {
+  start: Date | number,
+  end: Date | number,
+  timePeriodsKey?: string
+}
+
+export interface TimePeriod {
+  key: string // unique identifier
+  display: string
+  timeframeText: string
+  timeframeLength: () => string
+  start: () => Date
+  end: () => Date
+}
+
+export interface TimeFrameSection {
+  section: string
+  values: TimePeriod[]
+}
+
+export interface DateTimePickerState {
+  abbreviatedDisplay: string
+  fullRangeDisplay?: string
+  hidePopover: boolean
+  selectedRange: TimeRange
+  selectedTimeframe: TimePeriod
+  tabName: string
+}
+
+export interface CSSProperties {
+  [key: `${string}`]: string
+}
+
+export default defineComponent({
+  name: 'KDateTimePicker',
+  components: {
+    KButton,
+    KIcon,
+    KPop,
+    KSegmentedControl,
+    DatePicker,
+  },
+  props: {
+    icon: {
+      type: Boolean,
+      required: false,
+      default: true,
+    },
+    modelValue: {
+      type: Object as PropType<TimeRange> | undefined,
```

Comment:
I'm like 99% sure it's not valid as-is; needs to look like this I think? I'm assuming it's not required; however, then it needs a default value.

```suggestion
      type: Object as PropType<TimeRange | undefined>,
      required: false,
```